EU4toV2/Source/V2World/Province/Province.cpp

```diff
@@ -120,9 +122,15 @@ void V2::Province::convertFromOldProvince(const std::vector<std::shared_ptr<EU4:
 
 	// Single HRE province is enough
 	for (const auto& oldProvince: provinceSources)
+	{
 		if (oldProvince->inHre())
+		{
 			inHRE = true;
-
+			if (!hreDecentralized)
+				addCore("HRE");
```

Comment:
So, this is interesting. If we have a shattered byz, we'll be slapping HRE over the clay, not BYZ. I suppose there's no way around this as hre/byz/etc aren't union tags. Ok, leave this as is.

Reply:
HRE core adding used to be done in outProvince. I just moved it here, and blocked it if hreDecentralized

Reply:
Don't do that. out* functions are exclusively used for output, they should never do any logic.

Reply:
(if you fixed the logic by moving it here that's is good!)